routing/payment_lifecycle.go

```diff
@@ -186,6 +186,19 @@ func (p *paymentLifecycle) resumePayment() ([32]byte, *route.Route, error) {
 
 }
 
+// errorToPaymentFailure takes a path finding error and converts it into a
+// payment-level failure.
+func errorToPaymentFailure(err error) channeldb.FailureReason {
+	switch err {
+	case errNoTlvPayload, errNoPathFound, errMaxHopsExceeded,
```

Comment:
`errNoTlvPayload` isn't necessarily a path issue, but maybe that should be addressed in a different pr?

Reply:
Yes, I left that part unchanged. Just made it more obvious that that is the current logic. But agree that we should differentiate more. Another failure that we could expose is not having any info on the node at all (not in graph, not in the hints).